core/iwasm/compilation/aot_emit_stringref.c

```diff
@@ -0,0 +1,1452 @@
+                                EXCE_FAILED_TO_ENCODE_STRING, true, cmp,       \
+                                check_string_encode_succ))                     \
+            goto fail;                                                         \
+    } while (0)
+
+static bool
+aot_call_wasm_stringref_obj_new(AOTCompContext *comp_ctx,
+                                AOTFuncContext *func_ctx, LLVMValueRef str_obj,
+                                uint32 stringref_type, uint32 pos,
+                                LLVMValueRef *stringref_obj)
+{
+    LLVMValueRef param_values[3], func, value, res;
+    LLVMTypeRef param_types[3], ret_type, func_type, func_ptr_type;
+    uint32 argc = 2;
+
+    param_types[0] = INT8_PTR_TYPE;
+    param_types[1] = INT8_PTR_TYPE;
+    param_types[2] = I32_TYPE;
+    ret_type = INT8_PTR_TYPE;
+
+    if (stringref_type == WASM_TYPE_STRINGREF) {
+        GET_AOT_FUNCTION(wasm_stringref_obj_new, argc);
+    }
+    else if (stringref_type == WASM_TYPE_STRINGVIEWWTF8) {
+        GET_AOT_FUNCTION(wasm_stringview_wtf8_obj_new, argc);
+    }
+    else if (stringref_type == WASM_TYPE_STRINGVIEWWTF16) {
+        GET_AOT_FUNCTION(wasm_stringview_wtf16_obj_new, argc);
+    }
+    else {
+        argc = 3;
+        GET_AOT_FUNCTION(wasm_stringview_iter_obj_new, argc);
+    }
+
+    param_values[0] = func_ctx->exec_env;
+    param_values[1] = str_obj;
+    if (stringref_type == WASM_TYPE_STRINGVIEWITER) {
+        param_values[2] = I32_CONST(pos);
+    }
+
+    if (!(res = LLVMBuildCall2(comp_ctx->builder, func_type, func, param_values,
+                               argc, "create_stringref"))) {
+        aot_set_last_error("llvm build call failed.");
+        goto fail;
+    }
+
+    *stringref_obj = res;
+
+    return true;
+fail:
+    return false;
+}
+
+static LLVMValueRef
+aot_stringref_obj_get_value(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+                            LLVMValueRef stringref_obj)
+{
+    LLVMValueRef str_obj_ptr, str_obj, host_ptr_offset;
+
+    /* header */
+    host_ptr_offset = I32_CONST(comp_ctx->pointer_size);
+
+    if (!(stringref_obj =
+              LLVMBuildBitCast(comp_ctx->builder, stringref_obj, INT8_PTR_TYPE,
+                               "stringref_obj_i8p"))) {
+        aot_set_last_error("llvm build bitcast failed.");
+        goto fail;
+    }
+
+    if (!(str_obj_ptr =
+              LLVMBuildInBoundsGEP2(comp_ctx->builder, INT8_TYPE, stringref_obj,
+                                    &host_ptr_offset, 1, "str_obj_i8p"))) {
+        aot_set_last_error("llvm build gep failed.");
+        goto fail;
+    }
+
+    if (!(str_obj_ptr = LLVMBuildBitCast(comp_ctx->builder, str_obj_ptr,
+                                         GC_REF_PTR_TYPE, "str_obj_gcref_p"))) {
+        aot_set_last_error("llvm build bitcast failed.");
+        goto fail;
+    }
+
+    if (!(str_obj = LLVMBuildLoad2(comp_ctx->builder, GC_REF_TYPE, str_obj_ptr,
```

Comment:
Not sure whether the data is stored at n-byte aligned address, had better add `LLVMSetAlignment` for LLVMBuildLoad2 and LLVMBuildStore?

Reply:
The store is finished in runtime APIs, it should be 4bytes aligned, I've add the `LLVMSetAlignment`
Thanks